### PR TITLE
COTCO-144: architectural changes

### PR DESCRIPTION
**Pull request by robertbutacu**

### CODE REVIEW

File: app/uk/gov/hmrc/transitmovementapi/services/TransitService.scala [hunk 1]
```diff
@@ -19,27 +19,21 @@ package uk.gov.hmrc.transitmovementapi.services
...
 
 @Singleton
-class TransitService @Inject()(transitRepository: TransitRepository,
-                               crossingRepository: CrossingRepository,
+class TransitService @Inject()(ctcConnector: CtcConnector,
                                val auditConnector: AuditConnector)
                               (implicit val ec: ExecutionContext) extends AuditEvents {
 
-  def submitTransits(crossingId: String, transits: List[TransitSubmission])(implicit hc: HeaderCarrier): Future[Unit] = {
+  def submitTransits(transit: TransitSubmission)(implicit hc: HeaderCarrier): Future[Unit] = {
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/transitmovementapi/services/TransitService.scala]: Can we rename this to 'submitTransit'

File: conf/application.conf [hunk 2]
```diff
@@ -75,6 +75,8 @@ auditing {
 
 api.access.white-list.applicationIds = []
 
+with-backend = false
```
Comment (peter-hazell) [inline conf/application.conf]: could we make this config value a tad more descriptive, something like 'ctc-stub-enabled' = true / false

File: app/uk/gov/hmrc/transitmovementapi/services/test/TestService.scala [hunk 3]
```diff
@@ -17,19 +17,17 @@
...
-class TestService @Inject()(crossingRepository: CrossingRepository, transitRepository: TransitRepository)(implicit ec: ExecutionContext) {
+class TestService @Inject()(ctcConnector: CtcConnector)(implicit ec: ExecutionContext) {
 
   def clearAll(implicit hc: HeaderCarrier): Future[Unit] = {
-    for {
-      _ <- crossingRepository.clear()
-      _ <- transitRepository.clear()
-    } yield ()
+    //TODO implement
+    Future.successful(())
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/transitmovementapi/services/test/TestService.scala]: the test only endpoint that clears data in the backend service is still there, so should be able to implement this

File: app/uk/gov/hmrc/transitmovementapi/helpers/TransitEvent.scala [hunk 4]
```diff
@@ -14,22 +14,23 @@
...
+                        mrn: MovementReferenceNumber)
 
-object CrossingSubmission {
-  implicit val format: OFormat[CrossingSubmission] = Json.format[CrossingSubmission]
-}
+object TransitEvent {
+  def fromSubmission(transit: TransitSubmission): TransitEvent = TransitEvent(
+    userId = transit.userId,
+    deviceId = transit.deviceId,
+    mrn = transit.movementReferenceNumber
```
Comment (peter-hazell) [inline app/uk/gov/hmrc/transitmovementapi/helpers/TransitEvent.scala]: Some extra stuff to capture here but don't let it block merging this, something to update later

File: app/uk/gov/hmrc/transitmovementapi/services/test/TestService.scala (hunk 3 above)
Comment (robertbutacu) [reply app/uk/gov/hmrc/transitmovementapi/services/test/TestService.scala]: this is no longer needed as we are using wiremock

**Pull request merged**